source/agora/node/Ledger.d

```diff
@@ -414,25 +419,15 @@ public class Ledger
             ManagedDatabase.commitBatch();
         }
 
-        this.block_stats.setMetricTo!"agora_block_enrollments_gauge"(
-            this.enroll_man.validatorCount());
-        // there was a change in the active validator set
-        bool validators_changed = block.header.enrollments.length > 0
-            || this.enroll_man.validatorCount() != old_count;
-
-        // read back and cache the last block
-        if (externalized)
-            this.last_block = this.storage.readLastBlock();
-
         // Prepare maps for next block with maybe new enrollments
         log.trace("Storing active validators for next block using height {}.", block.header.height);
         this.enroll_man.updateValidatorIndexMaps(Height(block.header.height + 1));
 
         // Clear the unknown TXs every round (clear() is not @safe)
         this.unknown_txs = Set!Hash.init;
 
-        if (externalized && this.onAcceptedBlock !is null)
-            this.onAcceptedBlock(block, validators_changed);
+        // Update the known "last block"
+        this.last_block = deserializeFull!Block(serializeFull(block));
```

Comment:
This is the only part that isn't quite clean, because we need to have a mutable `block` but we get a `const` one.
Planning on solving this later, but block externalization being a bit more expensive isn't a big deal.